Add Recipe for Using Tailwind to Style Rendered Markdown (#5272)
withastro/docs: src/content/docs/en/recipes/tailwind-rendered-markdown.mdx

This recipe will teach you how to create a reusable Astro component to style your Markdown content using Tailwind's utility classes.

## Prerequisites

An Astro project that:  

	- has [Astro's Tailwind integration](/en/guides/integrations-guide/tailwind/) installed.  
	- uses Astro's [content collections](/en/guides/content-collections/).

## Setting Up `@tailwindcss/typography`

First, install `@tailwindcss/typography` using your preferred package manager.

<PackageManagerTabs>
 	<Fragment slot="npm">
	```shell 
	npm install -D @tailwindcss/typography
	```
	</Fragment>
  	<Fragment slot="pnpm">
	```shell 
	pnpm install -D @tailwindcss/typography
	```
	</Fragment>
  	<Fragment slot="yarn">
	```shell
	yarn add --dev @tailwindcss/typography
	```
	</Fragment>
</PackageManagerTabs>

Then, add the package as a plugin in your Tailwind configuration file.

```diff lang="js"
// tailwind.config.js
/** @type {import('tailwindcss').Config} */

export default {
  theme: {
    // ...
  },
  plugins: [
+   require('@tailwindcss/typography'),
    // ...
  ],
}
```

## Recipe

> 1. Create a `<Prose /> ` component to provide a wrapping `<div>` with a `<slot />` for your rendered Markdown. Add the style class `prose` alongside any desired **[Tailwind element modifiers](#how-to-use-element-modifiers)** in the parent element.   

Reviewer: ```suggestion
1. Create a `<Prose /> ` component to provide a wrapping `<div>` with a `<slot />` for your rendered Markdown. Add the style class `prose` alongside any desired [Tailwind element modifiers](https://tailwindcss.com/docs/typography-plugin#element-modifiers) in the parent element.   
```

    ```astro title="src/components/Prose.astro"
    ---
    ---
    <div 
      class="prose dark:prose-invert 
      prose-h1:font-bold prose-h1:text-xl 
      prose-a:text-blue-600 prose-p:text-justify prose-img:rounded-xl 
      prose-headings:underline">
      <slot />
    </div>
    ```
    :::tip
    The `@tailwindcss/typography` plugin uses [**element modifiers**](https://tailwindcss.com/docs/typography-plugin#element-modifiers) to style child components of a container with the `prose` class. 

    These modifiers follow the following general syntax: 

      ```
      prose-[element]:class-to-apply
      ``` 

>     For example, `prose-h1:font-bold` gives all `<h1>` tags the `font-bold` Tailwind class.

Reviewer: ```suggestion
    For example, `prose-h1:font-bold` gives all `<h1>` tags the `font-bold` Tailwind class.
    :::
```
Not sure why this didn't break withhout the closing part?  :sweat_smile:

---
title: Style Rendered Markdown with Tailwind Typography
description: Learn how to use @tailwind/typography to style your rendered Markdown
i18nReady: true
type: recipe
---

import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';

You can use [Tailwind](https://tailwindcss.com)'s Typography plugin to style rendered Markdown from sources such as Astro's [**content collections**](/en/guides/content-collections/).

This recipe will teach you how to create a reusable Astro component to style your Markdown content using Tailwind's utility classes.

## Prerequisites

An Astro project that:  

	- has [Astro's Tailwind integration](/en/guides/integrations-guide/tailwind/) installed.  

> 	- uses Astro's [content collections](/en/guides/integrations-guide/content-collections/).

Reviewer: ```suggestion
	- uses Astro's [content collections](/en/guides/content-collections/).
```

## Setting Up `@tailwindcss/typography`

First, install `@tailwindcss/typography` using your preferred package manager.

<PackageManagerTabs>
 	<Fragment slot="npm">
	```shell 
	npm install -D @tailwindcss/typography
	```
	</Fragment>
  	<Fragment slot="pnpm">
	```shell 
	pnpm install -D @tailwindcss/typography
	```
	</Fragment>
  	<Fragment slot="yarn">
	```shell
	yarn add --dev @tailwindcss/typography
	```
	</Fragment>
</PackageManagerTabs>

Then, add the package as a plugin in your Tailwind configuration file.

```diff lang="js"
// tailwind.config.js
/** @type {import('tailwindcss').Config} */

> export default = {

Author: ```suggestion
export default {
```

type: recipe
---

import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';

You can use [Tailwind](https://tailwindcss.com)'s Typography plugin to style rendered Markdown from sources such as Astro's [**content collections**](/en/guides/content-collections/).
This recipe will teach you how do create a reusable Astro component to **make your Markdown shine!**

## Prerequisites

An Astro project that:  

	- is [initialized with Tailwind](/en/guides/integrations-guide/tailwind/),  
	- is set up to use **content collections** or another Markdown rendering system.

## Setting Up `@tailwindcss/typography`

First, install `@tailwindcss/typography` using your preferred package manager.

<PackageManagerTabs>
 	<Fragment slot="npm">
	```shell 
	npm install -D @tailwindcss/typography
	```
	</Fragment>
  	<Fragment slot="pnpm">
	```shell 
	pnpm install -D @tailwindcss/typography
	```
	</Fragment>
  	<Fragment slot="yarn">
	```shell
	yarn add --dev @tailwindcss/typography
	```
	</Fragment>
</PackageManagerTabs>

Then, add the package as a plugin in your Tailwind configuration file.

```diff lang="js"
// tailwind.config.js
/** @type {import('tailwindcss').Config} */

> module.exports = {

Reviewer 2: Replace with `export default defineConfig = {`

Author: I do believe that the proper ESM syntax for this export would be to just export an object like so:
```diff
/** @type {import('tailwindcss').Config} */
- module.exports = {
+ export default {
  // ...
}
```

I could definitely fix that, since the Tailwind plugin for Astro generates a config file using ESM like I show above :)

Reviewer: I included this in my editing pass. Please correct if needed!

## Recipe

1. Create a `<Prose>` component to encapsulate your rendered Markdown. Here, you can use `prose` alongside **element modifiers** in the parent element.   

	```astro title="src/components/Prose.astro"
	---
	---
	<div 
	  class="prose dark:prose-invert 
	  prose-h1:font-bold prose-h1:text-xl 
	  prose-a:text-blue-600 prose-p:text-justify prose-img:rounded-xl 
	  prose-headings:underline">
	  <slot />
	</div>
	```

2. Then, pass your rendered Markdown as a child for your `<Prose>` component.  If you are using **content collections**, your code might look like this:

	```astro
	// src/pages/index.astro
	---
	import Prose from "../components/Prose.astro";
	import Layout from "../layouts/Layout.astro";
	import { getEntry } from 'astro:content';

	const entry = await getEntry('collection', 'entry');
	const RenderedMarkdown = await entry.render();
	---
	<Layout>
	  <Prose>
			<RenderedMarkdown />
	  </Prose>
	</Layout>
	```

You can view a live example of this recipe <a href="https://stackblitz.com/edit/withastro-astro-scvyma?file=src%2Fpages%2Findex.astro" target="_blank">here on StackBlitz</a>.

## Resources

> [Tailwind Typography Documentation](https://tailwindcss.com/docs/typography-plugin)

Reviewer 3: ```suggestion
- [Tailwind Typography Documentation](https://tailwindcss.com/docs/typography-plugin)
```

Match resource list format with other documents
